ui/core/components/gear_picker/selector_modal.tsx

```diff
@@ -162,9 +163,9 @@ export default class SelectorModal extends BaseModal {
 						onEquip: (eventID, item) => {
 							const equippedItem = gearData.getEquippedItem();
 							if (equippedItem) {
-								gearData.equipItem(eventID, equippedItem.withItem(item));
+								gearData.equipItem(eventID, equippedItem.withItem(item).withChallengeMode(this.player.getChallengeModeEnabled()));
```

Comment:
It would be cleaner to have `withItem()` automatically copy over the `_challengeMode` flag.